libfreerdp/core/peer.c

```diff
@@ -811,6 +811,7 @@ BOOL freerdp_peer_context_new(freerdp_peer* client)
 	client->autodetect->context = context;
 	update_register_server_callbacks(client->update);
 	autodetect_register_server_callbacks(client->autodetect);
+	update_register_io_callbacks(rdp->update);
```

Comment:
This should not be necessary with the new API

Reply:
why ?, is it not same initialization sequence as for client ?